### ISSUE TITLE
Drop PHP minimum required version

### ISSUE THREAD

**Issue by GaryJones** [danieltj27/Dark-Mode#124]
In doing #123, I saw that the minimum required version of PHP is 5.4.

I ran `phpcs` with the `testVersion` set as `5.2`, and it didn't report any new violations.

As such, with the intent to get this merged into core, I would suggest dropping the minimum PHP requirement from 5.4 to 5.2 in the following places:

- `readme.txt` Requires PHP header.
- `.phpcs.xml.dist` `testVersion` value.
- `composer.json` `php` requirement.

~~No actual PHP code needs to be changed. This is just adjusting the _claimed_ compatibility.~~ As spotted below, one bit does need changing.

**Comment by danieltj27** [comment]
Thanks Gary. I believe it was setup like that as at some point there was `! empty()` causing a warning in PHP 5.3 and below but that got removed a while back so we could drop it now.

Unless you beat me to it, I'll get this changed.

**Comment by GaryJones** [comment]
Once #123 is merged (which adds two of the files I mentioned would need changing), then a PR for this can be done.

**Comment by sybrew** [comment]
FYI: `__DIR__` is PHP 5.3+

http://php.net/manual/en/language.constants.predefined.php

use `dirname( __FILE__ )` instead.

**Comment by GaryJones** [comment]
> FYI: `__DIR__` is PHP 5.3+

Absolutely correct. Not sure why my check before didn't pick it up, but it does now.

![screenshot 2018-09-17 11 01 58](https://user-images.githubusercontent.com/88371/45617083-24691900-ba69-11e8-81f1-c65239e537ed.png)

**Comment by danieltj27** [comment]
Thanks all, I've merged pull #123. I'll fix the 5.2 issues very soon.

**Comment by GaryJones** [comment]
As per the items in the first post here, you didn't catch all of the instances. The [`testVersion`](https://github.com/danieltj27/Dark-Mode/blob/90236dc3d60c559c3a29209c0057aabcfb95a8d1/.phpcs.xml.dist#L26) needs updating as well.

**Comment by danieltj27** [comment]
[See #151 for more info](https://github.com/danieltj27/Dark-Mode/issues/151).
